### ISSUE TITLE
Undefined index: class at /media/psf/nextcloud/apps/files_accesscontrol/lib/Operation.php#88	

### ISSUE THREAD

**Issue by LukasReschke** [nextcloud/files_accesscontrol#17]
Undefined index: class at /media/psf/nextcloud/apps/files_accesscontrol/lib/Operation.php#88	

----

In todays build.

cc @nickvergessen

**Comment by MorrisJobke** [comment]
I just debugged a bit. It happens when a preview is requested (I logged $step in line 88):

```
{"reqId":"AQHFegrgieeRCHDNRT33","remoteAddr":"::1","app":"no app in context","message":"Exception: {\"Exception\":\"Exception\",\"Message\":\"Array
(
    [file] => lib/private/Route/Router.php
    [line] => 298
    [function] => call_user_func
    [args] => Array
        (
            [0] => \\u0000lambda_974
            [1] => Array
                (
                    [_route] => core_ajax_preview
                )
                
        )
        
        )
\",\"Code\":0,\"Trace\":\"#0 apps/files_accesscontrolÏ/libÏ/Operation.php(45): OCAÏFilesAccessControlÏOperation->isCreatingSkeletonFiles()
#1 apps/files_accesscontrol/lib/StorageWrapper.php(47): OCAÏFilesAccessControlÏOperation->checkFileAccess(Object(OCAÏFilesAccessControlÏStorageWrapper), 'thumbnails/20/3...')
#2 apps/files_accesscontrol/lib/StorageWrapper.php(242): OCAÏFilesAccessControlÏStorageWrapper->checkFileAccess('thumbnails/20/3...')
#3 lib/private/Files/View.php(1118): OCAÏFilesAccessControlÏStorageWrapper->file_put_contents('thumbnails/20/3...', 'Ïx89PNGÏrÏnÏx1AÏnÏx00Ïx00Ïx00ÏrIHD...')
#4 lib/private/Files/View.php(672): OCÏFilesÏView->basicOperation('file_put_conten...', 'thumbnails/20/3...', Array, '\x89PNG\r\n\x1A\n\x00\x00\x00\rIHD...')
#5 lib/private/Preview.php(1055): OC\Files\View->file_put_contents('thumbnails/20/3...', '\x89PNG\r\n\x1A\n\x00\x00\x00\rIHD...')
#6 lib/private/Preview.php(901): OC\Preview->storePreview(20, 32, 32)
#7 lib/private/Preview.php(1169): OC\Preview->resizeAndStore(20)
#8 lib/private/Preview.php(777): OC\Preview->generatePreview(20)
#9 lib/private/Preview.php(802): OC\Preview->getPreview()
#10 core/ajax/preview.php(64): OC\Preview->showPreview()
#11 lib/private/Route/Route.php(155) : runtime-created function(1): require_once('/Users/morrisjo...')
#12 [internal function]: __lambda_func(Array)
#13 lib/private/Route/Router.php(298): call_user_func('\x00lambda_974', Array)
#14 lib/base.php(969): OC\Route\Router->match('/core/preview.p...')
#15 index.php(48): OC::handleRequest()
#16 {main}\",\"File\":\"apps/files_accesscontrol/lib/Operation.php\",\"Line\":89}","level":3,"time":"2016-08-17T20:07:49+00:00","method":"GET","url":"\/server\/core\/preview.png?file=%2FBildschirmfoto+2016-08-11+um+15.10.43.png&c=8e57c3bc143d278e778530379dcf50ec&x=32&y=32&forceIcon=0","user":"admin"}
{"reqId":"AQHFegrgieeRCHDNRT33","remoteAddr":"::1","app":"PHP","message":"Undefined index: class at \/Users\/morrisjobke\/Projects\/nextcloud\/server\/apps\/files_accesscontrol\/lib\/Operation.php#91","level":0,"time":"2016-08-17T20:07:49+00:00","method":"GET","url":"\/server\/core\/preview.png?file=%2FBildschirmfoto+2016-08-11+um+15.10.43.png&c=8e57c3bc143d278e778530379dcf50ec&x=32&y=32&forceIcon=0","user":"admin"}
```

It seems to be a lambda function that doesn't have a class 🙈

**Comment by MorrisJobke** [comment]
fix is in #19 and #20 (master and stable10)
